### ISSUE TITLE
Writing custom gauge metric to Stackdriver fails due to unequal start & end dates

### ISSUE THREAD

**Issue by LeoHexspoor** [istio/istio#13235]
**Describe the bug**
Writing a gauge metric to Stackdriver fails due to to the start/end date of the metric being different. Mixer reports the following error:

> 2019-04-11T07:37:49.389691Z     error   adapters        Stackdriver returned: rpc error: code = InvalidArgument desc = Field timeSeries[0].points[0].interval.start_time had an invalid value of "2019-04-11T00:37:39.927086-07:00": The start time must be equal to the end time (2019-04-11T00:37:39.927087-07:00) for the gauge metric 'custom.googleapis.com/istio/example'.

Looking at the code I think this is due to the following two lines in `mixer/adapter/stackdriver/metric/metric.go`:
```go
start, _ := ptypes.TimestampProto(h.now())
end, _ := ptypes.TimestampProto(h.now())
```
which would always result in a slight difference between start and end date and it does not consider the metric type.

**Expected behavior**
A gauge metric to be successfully logged into Stackdriver.

**Steps to reproduce the bug**
Create a gauge type metric and try to log it to Stackdriver.

**Version**
Kubernetes: 1.12.6-gke.10
Istio: 1.0.6-gke.3

**Installation**
GKE managed addon

**Environment**
GKE

**Comment by bianpengyuan** [comment]
This issue should have already solved with 1.1 release: https://github.com/istio/istio/blob/master/mixer/adapter/stackdriver/metric/metric.go#L221. The next GKE addon release should have it.

**Comment by douglas-reid** [comment]
Closing this issue, as this has been resolved in the new releases.
